### ISSUE TITLE
W1b: `RandomAutoContrast.clip_output` is a no-op

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4436]
`RandomAutoContrast` takes a public `clip_output` constructor argument, documented at `kornia/augmentation/_2d/intensity/auto_contrast.py:31` as "if true clip output" and defaulting to `True`. `apply_transform` computes `normalize_min_max(input)` and then, at `:60`, clamps that into `[0, 1]` when the flag is set.

The clamp cannot do anything. `kornia.enhance.normalize_min_max` rescales each per-sample, per-channel plane onto `[0, 1]` by construction, so its output is already inside the interval the clamp would enforce. Setting the flag either way gives the identical tensor on a constant image, on an in-range image, on an image scaled into `[0, 2]` and on one shifted into `[-1, 0]` — the last two being the cases a caller would reach for the flag to protect against.

The argument therefore promises out-of-range protection that a user might rely on when feeding non-normalized data, and provides none, because there is nothing to protect against. Worth noting alongside it: on a constant image, the operator returns zeros for both settings, which is `normalize_min_max`'s behavior for a zero-range plane and is not what "auto contrast" suggests.

## Repro

```python
import torch, kornia.augmentation as K
from kornia.enhance import normalize_min_max

torch.manual_seed(0)
base = torch.rand(2, 3, 6, 8)
for tag, x in (("constant 0.5", torch.full((2, 3, 6, 8), 0.5)), ("[0,1] image", base),
               ("[0,2] image", base * 2.0), ("[-1,0] image", base - 1.0)):
    torch.manual_seed(0); t = K.RandomAutoContrast(clip_output=True, p=1.0)(x.clone())
    torch.manual_seed(0); f = K.RandomAutoContrast(clip_output=False, p=1.0)(x.clone())
    print(f"{tag:<13} clip_output=False min/max {f.min().item():.6g}/{f.max().item():.6g}   "
          f"max|clip_output=True - clip_output=False| = {(t - f).abs().max().item()}")
torch.manual_seed(0)
print("max|RandomAutoContrast(x) - kornia.enhance.normalize_min_max(x)|:",
      (K.RandomAutoContrast(p=1.0)(base.clone()) - normalize_min_max(base)).abs().max().item())
n = normalize_min_max(base * 2.0)
print("normalize_min_max already returns min/max:", round(n.min().item(), 6), round(n.max().item(), 6))
```

## Observed

```
constant 0.5  clip_output=False min/max 0/0   max|clip_output=True - clip_output=False| = 0.0
[0,1] image   clip_output=False min/max 0/0.999999   max|clip_output=True - clip_output=False| = 0.0
[0,2] image   clip_output=False min/max 0/1   max|clip_output=True - clip_output=False| = 0.0
[-1,0] image  clip_output=False min/max 0/0.999999   max|clip_output=True - clip_output=False| = 0.0
max|RandomAutoContrast(x) - kornia.enhance.normalize_min_max(x)|: 0.0
normalize_min_max already returns min/max: 0.0 1.0
```

## Expected

Two coherent outcomes:

1. **Remove the argument.** It is a public constructor parameter, so it needs a deprecation cycle: keep accepting it, warn, and drop it after the window. Callers passing it explicitly see no change in output at any point.
2. **Keep it and document it as a no-op** — a compatibility argument retained for signature stability, with the docstring at `auto_contrast.py:31` saying the output of `normalize_min_max` is already in `[0, 1]`, so the flag cannot change the result.

The audit recommends (1). A public argument that looks like a safety switch and is not one is worse than no argument at all, and the removal costs no caller any output change.

## Classification

**API completeness** — a public constructor argument with no observable effect on any input tested. Idiom: none.

## Disposition

`MERGE IN WINDOW — behavior change`. Removing a public constructor argument is an API change and belongs in the coordinated Repair & Deprecation window, even though no output moves. Batch 6 documents the current behavior as it is and pins it. Option (2), the docstring correction alone, is a focused fix and welcome as a PR in the meantime.

## Environment

kornia 0.9.0rc1 @ `d6a2c138` (main), torch 2.14.0, python 3.11, cpu. Found during the augmentation conventions audit (batch 6, tracking issue #4407); the batch documents the behavior above as it is and pins it, and does not change it.

Posted on behalf of @ducha-aiki by Claude (Fable 5.1).
